### PR TITLE
feat(agentFlow): Fix duplicate node and add test cases for delete and duplicate nodes

### PR DESCRIPTION
**Pull request by j-sanaa**
- Tests added for delete and duplicate node
-  Fixed duplicate node bug where 'node.data.id` wasn't updated during duplication
- Added fix for [[FLOWISE-257]:](https://jira2.workday.com/browse/FLOWISE-257) agentflow clear() and addNode() don't update canvas after bidirectional sync refactor
- Fixed lint warnings from #5827 

NOTE: Bug where duplicate nodes are dragged together (shown in recording) will be resolved in a separate PR

https://github.com/user-attachments/assets/b3f0c0a6-f101-4fe7-9333-fec15e74417c

### CODE REVIEW

File: packages/agentflow/src/infrastructure/store/AgentflowContext.test.tsx [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+        const duplicatedNode = result.current.state.nodes.find((n) => n.id.startsWith('node-1_copy_'))
+
+        // Extract timestamp from ID
+        const timestampMatch = duplicatedNode?.id.match(/_copy_(\d+)$/)
+        expect(timestampMatch).toBeTruthy()
+
+        const timestamp = parseInt(timestampMatch![1], 10)
+        expect(timestamp).toBeGreaterThanOrEqual(beforeTimestamp)
+        expect(timestamp).toBeLessThanOrEqual(afterTimestamp)
+    })
```
[the remark this answers is not in the document]
Comment (j-sanaa) [reply packages/agentflow/src/infrastructure/store/AgentflowContext.test.tsx]: No longer needed as date is replaced with sequential ids

File: packages/agentflow/src/infrastructure/store/AgentflowContext.tsx [hunk 2]
```diff
@@ -52,8 +54,8 @@ export function AgentflowStateProvider({ children, initialFlow }: AgentflowState
     })
 
     // Store ReactFlow local state setters in refs which are populated by AgentflowCanvas
-    const localNodesSetterRef = useRef<((nodes: FlowNode[]) => void) | null>(null)
-    const localEdgesSetterRef = useRef<((edges: FlowEdge[]) => void) | null>(null)
+    const localNodesSetterRef = useRef<((_nodes: FlowNode[]) => void) | null>(null)
```
Comment (jocelynlin-wd) [inline packages/agentflow/src/infrastructure/store/AgentflowContext.tsx]: nit, given the function type is defined multiple times in the file, we can simply make it a function type for this file and reuse the type while avoiding lint warnings. 

Example https://github.com/FlowiseAI/Flowise/blob/main/packages/agentflow/src/core/types/index.ts#L50 

note we don't need to add type to core but to the top of this file since they are only used here

File: packages/agentflow/src/infrastructure/store/AgentflowContext.test.tsx [hunk 3]
```diff
@@ -0,0 +1,836 @@
...
+
+// Wrapper component for testing
+const createWrapper = (initialFlow?: FlowData) => {
+    function Wrapper({ children }: { children: ReactNode }) {
+        return <AgentflowStateProvider initialFlow={initialFlow}>{children}</AgentflowStateProvider>
+    }
+    return Wrapper
+}
+
+describe('AgentflowContext - deleteNode', () => {
```
Comment (jocelynlin-wd) [inline packages/agentflow/src/infrastructure/store/AgentflowContext.test.tsx:21]: thanks for adding tests, do you mind also update `TESTS.md` and also `coverageThreshold`  in `jest.config.ts` ?

**Review by jocelynlin-wd — APPROVED** [review]

**Comment by jocelynlin-wd** [comment]
the issue with duplicate node shown in the recording (identical node name and nodes are dragged together etc) will be handled in a separate PR
